### ISSUE TITLE
Training on Chinese dataset does not work

### ISSUE THREAD

**Issue by xiaonengmiao** [SafeAILab/EAGLE#179]
Hello, thanks for this nice work.

I trained a eagle "English" verison model on qwen2.5, and tested it on mt_bench data. The ratio is about 3.0. I also tested it with `some chinese question task` the ratio droped to about 1.5. 

So I decided to train a "Chinese" version eagle model with a [`Chinese Sharegpt dataset`](https://huggingface.co/datasets/FreedomIntelligence/sharegpt-chinese). The data generation and training works similar to the "English" verison: after 200 epoch, train accuracy : 81.04% test accuracy 79.22% Loss: 0.7023.

But the problem is when i test it using the same `some chinese question task` above, there is no accrracy at all. The speed ratio droped blow 1.0. Does this mean the [`Chinese Sharegpt dataset`](https://huggingface.co/datasets/FreedomIntelligence/sharegpt-chinese) i used is not good? or I missed something? 

Thanks for your help

**Comment by xuu416** [comment]
May I take a look at the training code for Qwen2.5 Eagle?

**Comment by Liyuhui-12** [comment]
Your test accuracy is high, but the actual inference cannot be accelerated. A possible reason is that your training data (including the test set) differs significantly from the actual inference scenario.

**Comment by souyang11** [comment]
Hello, I am currently using sglang + yuhuili/EAGLE-Qwen2-7B-Instruct for inference, and I noticed that the throughput has slowed down. Do you know the reason for this? thank you.

**Comment by liusong1222** [comment]
@xiaonengmiao  Hello, I have same problem. Have you ever tried testing the acceptance rate directly using the training data?

**Comment by wenqf11** [comment]
> > I also trained qwen3-8b eagle3 using SpecForge on a Chinese dataset(Chinese-DeepSeek-R1-Distill-data-110k)(generated answer using Qwen3-8B), but after 10 epoch only got (Position 0) train acc 0.7. and test using vllm with prompts from training dataset, the acceptance rate is 25%~30%, speedup 1.3X~1.4X. Any advice, thanks!
> 
> I got same problem. Any advice, thanks!

Just make sure train acc(position 0) is higher around 0.8.( I just make learning rate decay as expected)
